airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java

```diff
@@ -159,6 +167,35 @@ public boolean isV2Destination() {
     return true;
   }
 
+  @Override
+  public void verifyCatalog(@NotNull ParsedCatalog parsedCatalog) {
+    super.verifyCatalog(parsedCatalog);
+
+    // Check for collisions between column names, after applying the 63-character truncation
+    List<String> streamErrorMessages = new ArrayList<>();
+    for (StreamConfig stream : parsedCatalog.getStreams()) {
+      Set<String> truncatedColumnNames = new HashSet<>();
```

Comment:
probably use a `Map<truncatedCol, originalCol>` so when we report the error, we actually report the possible collisions of the original column names for UX ?

Reply:
I thought about that, felt it was actually easier if we just gave them the common prefix instead of a list of all offending columns? Since the UI sorts field names alphabetically, it should be easy to find them

.... modulo replacing special characters with underscores I guess, idk

I mildly lean towards "don't generate a giant list of foo_bar, foo_baz, foo_whatever" when we could just tell the user "pick one column that starts with foo_", but don't feel strongly

Reply:
I don't feel strongly against it either. I'm more worried about users coming back asking what is this column name you are emitting in logs, we don't see them in column selection page.

Reply:
mmhm. I'm trying to clarify that via the error message, but maybe being more explicit would be good... e.g. `stream X.Y has multiple columns that start with "foo", please remove all but one`

Reply:
Yeah thats works too I guess.